lib/pool.js

```diff
@@ -201,23 +202,52 @@ function getConnection(a1, a2) {
 
 getConnectionPromisified = nodbUtil.promisify(getConnection);
 
-function close(closeCb) {
+function closeWrapper(mode, closeCb) {
   var self = this;
-
-  nodbUtil.assert(arguments.length === 1, 'NJS-009');
-  nodbUtil.assert(typeof closeCb === 'function', 'NJS-006', 1);
-
-  self._close(function(err) {
+  self._close({forceClose:mode}, function (err) {
     if (!err) {
       self._isValid = false;
 
       self.emit('_after_close', self);
     }
-
     closeCb(err);
   });
 }
 
+function close(a1, a2) {
+  var self = this;
+
+  nodbUtil.assert(arguments.length === 1 || arguments.length === 2, 'NJS-009');
+
+  const closeCb = arguments[arguments.length - 1];
+  nodbUtil.assert(typeof closeCb === 'function', 'NJS-006', 1);
+
+  const forceClose = arguments.length === 1 ? false : true;
+  self.closing = forceClose;
+
+  if (arguments.length === 1) {
+    self.closeWrapper(false, closeCb);
+  } else {
+    
+    //If a drainTime has been provided, it will use that value, given that connections count is not 0
+    //if connectionsOut is === 0, closePool is called right away, as there is no need to wait at all.
+    const drainTime = self._connectionsOut === 0 ? 0 : arguments[arguments.length - 2] * 1000;
+    //SetTimeout does not accept numbers greather than a 32bit signed integer.
+    const upperLimitTimeout = 2147483647;
+
+    nodbUtil.assert(typeof drainTime === 'number', 'NJS-006', 2);
+    nodbUtil.assert(drainTime >= 1 && drainTime <= upperLimitTimeout, 'NJS-005', 2);
```

Comment:
Shouldn't this allow drainTime of 0, meaning force the connection pool to close?

Reply:
Yeah, I meant to have that as 0, good catch.